tests/udp_socket.rs

```diff
@@ -343,9 +342,15 @@ fn reconnect_udp_socket_receiving() {
         vec![ExpectEvent::new(ID1, Interests::READABLE)],
     );
 
-    // Read only a part of the data.
-    let max = 4;
-    expect_read!(socket1.recv(&mut buf[..max]), &DATA2[..max]);
+    if cfg!(not(target_os = "windows")) {
+        // On all platform except Windows, read only a part of the data.
+        let max = 4;
+        expect_read!(socket1.recv(&mut buf[..max]), &DATA2[..max]);
+    }
+    if cfg!(target_os = "windows") {
+        // On Windows, read all data, reading part of data returns error WSAEMSGSIZE (10040)
```

Comment:
Let's not split up this test in Windows/others. Lets just read the entire packet on all platform.

Also lets add a note in the `read`/`peek`(`_from`) docs that on Windows reading with a too small a buffer will return an error. Encouraging people to use `1 << 16` sized buffers (the maximum size for a UDP packet).

Reply:
ok, will read all

Reply:
I've added note for all read/peek(_from)